data_model/src/expression.rs

```diff
@@ -24,6 +24,142 @@ use serde::{Deserialize, Serialize};
 
 use super::{query::QueryBox, Value, ValueBox};
 
+/// Generate expression structure and basic impls for it.
```

Comment:
The syntax is much better now. We can probably improve the syntax slightly more by having a derive macro, but that is well beyond the scope of this PR.